core/state/statedb.go

```diff
@@ -47,6 +47,15 @@ type revision struct {
 	journalIndex int
 }
 
+type objectState struct {
+	typ  int  // 0 => update; 1 => delete
```

Comment:
Shouldn't we use an enum for this? Or, if we know it's only two types, perhaps a boolean?

Reply:
I honestly dont know how to do that but i have an enum